lib/inventory_sync/async/inventory_self_host_sync.rb

```diff
@@ -0,0 +1,30 @@
+module InventorySync
+  module Async
+    class InventorySelfHostSync < QueryInventoryJob
+      set_callback :step, :around, :create_facets
+
+      def create_facets
+        # get the results from the event
+        results = yield
+
+        add_missing_insights_facet(results.uuid_by_fqdn) unless results.uuid_by_fqdn.empty?
+        results
+      end
+
+      private
+
+      def add_missing_insights_facet(uuids_hash)
+        facet = InsightsFacet.find_or_create_by(host_id: ForemanRhCloud.foreman_host.id) do |facet|
+          facet.uuid = uuids_hash.values.first
+        end
+
+        # fix empty uuid in case the facet already exists
+        facet.update(uuid: uuids_hash.values.first) unless facet.uuid
+      end
+
+      def inventory_url
```

Comment:
should it be `request_url`?
iiuc it should override the method in `query_inventory_job.rb` ?

Reply:
Nice catch!